File: backend/src/main/java/com/audiostory/model/Story.java
package com.audiostory.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "stories")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Story {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private String author;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;
    
    @OneToMany(mappedBy = "story", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("sortOrder ASC")
    @Builder.Default
    private List<StoryImage> images = new ArrayList<>();
    
    @OneToMany(mappedBy = "story", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("episodeNumber ASC")
    @Builder.Default
    private List<Episode> episodes = new ArrayList<>();
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/lib/models/story.dart
import 'episode.dart';

class Story {
  final int id;
  final String title;
  final String? description;
  final String? author;
  final int? categoryId;
  final String? categoryName;
  final List<String> images;
  final List<Episode> episodes;
  final int episodeCount;

  Story({
    required this.id,
    required this.title,
    this.description,
    this.author,
    this.categoryId,
    this.categoryName,
    this.images = const [],
    this.episodes = const [],
    this.episodeCount = 0,
  });

  factory Story.fromJson(Map<String, dynamic> json) {
    return Story(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      author: json['author'],
      categoryId: json['categoryId'],
      categoryName: json['categoryName'],
      images: json['images'] != null 
          ? List<String>.from(json['images']) 
          : [],
      episodes: json['episodes'] != null
          ? (json['episodes'] as List).map((e) => Episode.fromJson(e)).toList()
          : [],
      episodeCount: json['episodeCount'] ?? 0,
    );
  }

  String get coverImage => images.isNotEmpty ? images.first : '';
}
